site/app/controllers/admin/AutogradingConfigController.php

```diff
@@ -232,14 +233,47 @@ public function configUsedBy($config_path = null) {
     }
 
     /**
-     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}", methods={"GET"})
+     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}/{mode}", methods={"GET"})
      * @param string $g_id Gradeable ID
+     * @param string $mode The mode notebook builder should open in.  May be either 'new' or 'edit', this lets
+     * notebook builder know to save a new configuration or edit the existing one
      * @AccessControl(role="INSTRUCTOR")
      */
-    public function notebookBuilder(string $g_id) {
-        $gradeable = $this->core->getQueries()->getGradeableConfig($g_id);
+    public function notebookBuilder(string $g_id, string $mode) {
+        $gradeable = null;
+        try {
+            $gradeable = $this->core->getQueries()->getGradeableConfig($g_id);
+        }
+        catch (\Exception $exception) {
+            // Pass
```

Comment:
We should not be passing here, rather just explicitly make a redirect back to `$this->core->buildCourseUrl()` or something, no point going forward if we don't have a gradeable given it's required below.

Reply:
Changed